server/resp/src/test/java/org/infinispan/server/resp/RespBlockingCommandsTest.java

```diff
@@ -0,0 +1,146 @@
+import org.infinispan.server.resp.test.RespTestingUtil;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import io.lettuce.core.KeyValue;
+import io.lettuce.core.RedisCommandExecutionException;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.api.sync.RedisCommands;
+
+/**
+ * Class for blocking commands tests.
+ *
+ * @author Vittorio Rigamonti
+ * @since 15.0
+ */
+@Test(groups = "functional", testName = "server.resp.RespBlockingCommandsTest")
+public class RespBlockingCommandsTest extends SingleNodeRespBaseTest {
+
+   private CacheMode cacheMode = CacheMode.LOCAL;
+   private boolean simpleCache;
+
+   @Factory
+   public Object[] factory() {
+      return new Object[] {
+            new RespBlockingCommandsTest(),
+            new RespBlockingCommandsTest().simpleCache()
+      };
+   }
+
+   RespBlockingCommandsTest simpleCache() {
+      this.cacheMode = CacheMode.LOCAL;
+      this.simpleCache = true;
+      return this;
+   }
+
+   @Override
+   protected String parameters() {
+      return "[simpleCache=" + simpleCache + ", cacheMode=" + cacheMode + "]";
+   }
+
+   @Override
+   protected void amendConfiguration(ConfigurationBuilder configurationBuilder) {
+      if (simpleCache) {
+         configurationBuilder.clustering().cacheMode(CacheMode.LOCAL).simpleCache(true);
+      } else {
+         configurationBuilder.clustering().cacheMode(cacheMode);
+      }
+   }
+
+   @Test
+   void testBlpopAsync() throws InterruptedException, ExecutionException, TimeoutException {
+      RedisAsyncCommands<String, String> redis = redisConnection.async();
+      var client = createClient(30000, server.getPort());
+      RedisAsyncCommands<String, String> redis1 = client.connect().async();
+      redis1.lpush("keyY", "firstY").get();
+      redis.rpush("key1", "first", "second", "third").get();
+      var res = redis.blpop(0, "keyZ");
+      Thread.sleep(1000);
+      redis1.lpush("keyZ", "firstZ").get();
+      try {
+         var response = res.get(4, TimeUnit.SECONDS);
+         assertThat(response.getKey()).isEqualTo("keyZ");
+         assertThat(response.getValue()).isEqualTo("firstZ");
+      } finally {
+         RespTestingUtil.killClient(client);
+      }
+   }
+
+   @Test
+   public void testBlpop() throws InterruptedException, ExecutionException {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      var client = createClient(30000, server.getPort());
+      RedisCommands<String, String> redisBlock = client.connect().sync();
+      redis.rpush("key1", "first", "second", "third");
+      CompletableFuture<KeyValue<String, String>> cf = new CompletableFuture<>();
+      KeyValue<String, String> res;
+      var tpool = Executors.newCachedThreadPool();
+      tpool.submit(() -> {
+         cf.complete(redisBlock.blpop(0, "keyZ"));
+         return null;
+      });
+      // Ensure lpush is after blpop
```

Comment:
I guess we can only ensure that using the async client and a latch, not the test would fail. Like:

```java
      CountDownLatch latch = new CountDownLatch(1);
      RedisAsyncCommands<String, String> redisBlock = client.connect().async();
      CompletableFuture<KeyValue<String, String>> cf = CompletableFuture.supplyAsync(() -> {
         RedisFuture<KeyValue<String, String>> rf = redisBlock.blpop(0, "keyZ");
         latch.countDown();
         return rf.toCompletableFuture();
      }, testExecutor()).thenCompose(v -> v);
      // Ensure lpush is after blpop
      assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
```

This should cover the above test, too.

Reply:
nice!  Implemented, thanks!